src/conv/invokers/mlir_impl_gemm.cpp

```diff
@@ -106,7 +103,7 @@ InvokerFactory MakeMlirFwdInvokerFactory(const ConvolutionContext& ctx)
     MlirConvArgs args =
         makeMlirConvArgs(in_dims, in_strides, weights_dims, weights_strides, out_dims, out_strides);
 
-    return [=](const std::vector<Kernel>& kernels) {
+    return [=](const std::vector<Kernel>& kernels) mutable {
```

Comment:
Could you please clarify why g++ wants `mutable`. If this is a g++ specific quirk, let's make it so:
```C++
/// Workaround for a g++ specific quirk. When the issue is fixed in g++, we can insert here something like
/// "&& __GNUC__ >7 && __GNUC_MINOR__ < 6" (just an example).
#if defined(__GNUC__)
#define GNUC_MUTABLE mutable
#else
#define GNUC_MUTABLE
#endif
...
        return [=](const std::vector<Kernel>& kernels) GNUC_MUTABLE {
```

Reply:
@atamazov As far as I can tell this is not a quirk and clang should complain too.

A few lines below I have:

```cpp
args.filter.basePtr = filter;
args.filter.data = filter;
...
```

lambda pass-by value are implicitly `const`. In this case, the captured `args` should be a read-only variable. I need to add this mutable just so that it is allowed to update the captured `args`.

Reply:
True, thanks for explanation.